kg_microbe/transform_utils/uniprot/uniprot.py

```diff
@@ -0,0 +1,178 @@
+        input_dir = str(self.input_base_dir)+"/"+self.source_name 
+        #Get all organisms downloaded into raw directory
+        ncbi_organisms = []
+        for f in os.listdir(input_dir):
+            if f.endswith(".json"):
+                ncbi_organisms.append(f.split('.json')[0])
+
+        # make directory in data/transformed
+        os.makedirs(self.output_dir, exist_ok=True)
+
+        with open(self.output_node_file, "w") as node, open(
+            self.output_edge_file, "w"
+        ) as edge:
+            node_writer = csv.writer(node, delimiter="\t")
+            node_writer.writerow(self.node_header)
+            edge_writer = csv.writer(edge, delimiter="\t")
+            edge_writer.writerow(self.edge_header)
+
+            #Generates __enz_data
+            self.add_org_to_enz(input_dir, ncbi_organisms, 'uniprot', node_writer, edge_writer)
+
+
+        drop_duplicates(self.output_node_file)
+        drop_duplicates(self.output_edge_file)
+        
+    #Takes ncbitaxon ids as dict in kgx format
+    def add_org_to_enz(self, input_dir, nodes, source, node_writer, edge_writer):
+        '''Submit data to the graph.'''
+
+
+        # Create Organism and Enzyme nodes:
+        self.get_uniprot_values_from_file(input_dir, nodes, source, node_writer, edge_writer)
+
+    def get_uniprot_values_from_file(self,input_dir, nodes, source, node_writer, edge_writer):
+
+        with tqdm(total=len(nodes) + 1, desc="Processing files") as progress:
+            for i in tqdm(range(len(nodes))):
+                org_file = input_dir + '/' + nodes[i] + ".json"
+                if not os.path.exists(org_file):
+                    print('File does not exist: ',org_file,', exiting.')
+                    sys.exit()
+
+                else:
+                    with open(org_file, encoding='utf-8') as json_file:
+                        values = json.load(json_file)
+                        self.write_to_df(values, edge_writer, node_writer)
+
+                progress.set_description(f"Processing Uniprot File: {nodes[i]}.yaml")
+                # After each iteration, call the update method to advance the progress bar.
+                progress.update()
+
+
+    def get_uniprot_values_organism(self,organism_ids, fields, keywords,edge_writer, node_writer, batch_size, verbose=False, num_threads=0):
+        values = []
+
+        print('querying uniprot for enzymes per organism (' + str(len(organism_ids)) +  ') by batch size (' + str(batch_size) + ')')
+        for i in tqdm(range(0, len(organism_ids), batch_size)):
+            values = self.get_uniprot_values_from_file(organism_ids, i, batch_size, fields, keywords, values,verbose)
+            
+            self.write_to_df(values, edge_writer, node_writer)
+            print('wrote to dataframe')
+
+    
+    def parse_binding_site(self,binding_site_entry):
+
+        chem_list=re.findall(r'/ligand_id="ChEBI:(.*?)";',binding_site_entry)
```

Comment:
do we always expect binding sites to be stored under `ligand_id` and will they always have ChEBI IDs (e.g., will they ever have IDs for other chemical knowledgebases)?

Reply:
Uniprot only includes CHEBI id's for binding sites, so none will be excluded unless they are labelled "substrate" thus without any specific chemical information/mappings.